app/TaskRouter/WorkspaceFacade.php

```diff
@@ -3,104 +3,182 @@
-     * @param $property
-     * @return mixed
+     * Magic getter to lazy load subresources
+     *
+     * @param string $property Subresource to return
+     *
+     * @return \Twilio\ListResource The requested subresource
+     *
+     * @throws \Twilio\Exceptions\TwilioException For unknown subresources
      */
     public function __get($property)
     {
-        return $this->workspace->$property;
+        return $this->_workspace->$property;
     }
 
     /**
-     * Magic method to write the attributes into the wrapped workspace
-     * @param $property
-     * @param $value
-     * @return mixed
+     * Gets an activity instance by its friendly name
+     *
+     * @param $activityName Friendly name of the activity to search for
+     *
+     * @return ActivityInstance of the activity found or null
      */
-    public function __set($property, $value)
+    function findActivityByName($activityName)
     {
-        $this->workspace->$property = $value;
-        return $this;
+        $this->cacheActivitiesByName();
+        return $this->_activities[$activityName];
     }
 
     /**
-     * @param $activityName Name of the activity to search for
-     * @return the activity found or null
+     * Caches the activities in an associative array which links friendlyName with
+     * its ActivityInstance
      */
-    function findActivityByName($activityName)
+    protected function cacheActivitiesByName()
     {
-        foreach ($this->workspace->activities->read() as $activity) {
-            if ($activity->friendlyName === $activityName)
-                return $activity;
+        if (!$this->_activities) {
+            $this->_activities = array();
+            foreach ($this->_workspace->activities->read() as $activity) {
+                $this->_activities[$activity->friendlyName] = $activity;
+            }
         }
     }
 
     /**
-     * @param $taskQueueName Name of the task queue to search for
+     * Looks for a worker by its SID
+     *
+     * @param $sid string with the Worker SID
+     *
+     * @return mixed worker found or null
+     */
+    function findWorkerBySid($sid)
+    {
+        return $this->_workspace->workers->getContext($sid);
+    }
+
+    /**
+     * Returns an associative array with
+     *
+     * @return mixed array with the relation phone -> workerSid
+     */
+    function getWorkerPhones()
+    {
+        $worker_phones = array();
+        foreach ($this->_workspace->workers->read() as $worker) {
+            $workerAttribs = json_decode($worker->attributes);
+            $worker_phones[$workerAttribs->contact_uri] = $worker->sid;
+        }
+        return $worker_phones;
+    }
+
+    /**
+     * Looks for a Task Queue by its friendly name
+     *
+     * @param $taskQueueName string with the friendly name of the task queue to
+     * search for
+     *
      * @return the activity found or null
      */
     function findTaskQueueByName($taskQueueName)
```

Comment:
Here you can also filter taskqueues by friendly name instead of going through all of them